src/routes/Bridge.js

```diff
@@ -0,0 +1,676 @@
+
+        // Access the decentralized web!
+
+        // Legacy providers may only have ethereum.sendAsync
+        const accounts = await provider.request({
+          method: 'eth_accounts'
+        })
+        console.log(accounts, provider)
+        let Web3Provider = await Sdk.MetaMaskWalletProvider.connect()
+        this.sdk = await new Sdk.Sdk(Web3Provider, {
+          networkName: 'mainnet',
+          omitWalletProviderNetworkCheck: true
+        })
+        await this.sdk.computeContractAccount()
+        console.info('SDK created')
+        const { state: sdkState } = this.sdk
+        const { accountAddress: ContractedAccount } = sdkState
+        this.contract_account = ContractedAccount
+        this.wallet_address = sdkState.walletAddress
+        let acc_bal = await this.sdk.getAccountBalances({
+          tokens: ['0x6b175474e89094c44da98b954eedeac495271d0f']
+        })
+        this.balance = ethers.utils.formatEther(
+          acc_bal.items[1] == null ? 0 : acc_bal.items[1].balance
+        )
+        console.log(
+          'balances',
+          ethers.utils.formatEther(
+            acc_bal.items[1] == null ? 0 : acc_bal.items[1].balance
+          )
+        )
+        this.sokol_sdk = await new Sdk.Sdk(Web3Provider, {
+          networkName: 'xdai',
+          omitWalletProviderNetworkCheck: true
+        })
+        await this.sokol_sdk.computeContractAccount()
+        let sokol_acc_bal = await this.sokol_sdk.getAccountBalances()
+        console.log(sokol_acc_bal)
+        this.sokol_bal = ethers.utils.formatEther(
+          sokol_acc_bal.items[0] == null ? 0 : sokol_acc_bal.items[0].balance
+        )
+        await this.setState({
+          unlock: true,
+          loading: false
+        })
+        console.log(this.unlock)
+      } else {
+        // if the provider is not detected, detectEthereumProvider resolves to null
+        await this.setState({
+          loading: false
+        })
+        alert('Please install MetaMask!')
+        console.error('Please install MetaMask!')
+      }
+    } catch (err) {
+      console.log(err)
+      this.setState({
+        loading: false,
+        loading_msg: ''
+      })
+      alert('Something went wrong! Please try again')
+      return null
+    }
+  }
+
+  getBalance = async () => {
+    await this.setState({
+      loading: true
+    })
+    let acc_bal = await this.sdk.getAccountBalances({
+      tokens: ['0x6b175474e89094c44da98b954eedeac495271d0f']
+    })
+    this.balance = ethers.utils.formatEther(
+      acc_bal.items[1] == null ? 0 : acc_bal.items[1].balance
+    )
+    let sokol_acc_bal = await this.sokol_sdk.getAccountBalances()
+    console.log(sokol_acc_bal)
+    this.sokol_bal = ethers.utils.formatEther(
+      sokol_acc_bal.items[0] == null ? 0 : sokol_acc_bal.items[0].balance
+    )
+    // alert("Balances updated");
+    await this.setState({
+      loading: false
+    })
+  }
+
+  xDai_dai = async () => {
```

Comment:
This is a huge logic. can you move to a different file/class and make it reusable?